src/ProductService.php

```diff
@@ -382,6 +582,36 @@ public function buildProducts($entity_type, $entity_id, $fieldname) {
     // Filter invalid products.
     $product_data = array_filter($product_data);
 
+    // Replace unavailable products with ones from the search term fallback.
+    $replace = [];
+    foreach ($product_data as $asin => $data) {
+      if (!$data['product_available']) {
+        $replace[] = $asin;
+      }
+    }
+
+    if (!empty($replace)) {
+      $fallback_asins = $this->getSearchResults($search_terms, ProductService::AMAZON_CATEGORY_DEFAULT);
+      $fallback_data = $this->getProductData($fallback_asins);
+      // Replace outdated products and keep the result order: $fallback_asins
+      // contains ordered results (top first).
+      $product_data = array_diff_key($product_data, array_flip($replace));
+      foreach ($fallback_asins as $asin) {
+        if (
+          empty($product_data[$asin])
+          && !empty($fallback_data[$asin])
+          && $fallback_data[$asin]['product_available']
+          && !empty($replace)
```

Comment:
loop breaks as soon as this variable is empty and it is checked whether it is empty before, so it should never be empty at this point